CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -587,12 +593,15 @@ function Staff:increaseWage(amount)
   local wage_raised = true
   if self.profile.wage >= max_salary then
     wage_raised = false -- Already at max salary
-  elseif self.profile.wage + amount > max_salary then
-    -- Maximum salary hit. The staff member will never be unhappy again
-    self.profile.wage = max_salary
   else
+    local new_wage = self.profile.wage + amount
+    if self.profile.wage + amount > max_salary then
+      -- Maximum salary hit. The staff member will never be unhappy again
+      new_wage = max_salary
+    end
     -- Apply new salary
-    self.profile.wage = self.profile.wage + amount
+    self.profile.wage = new_wage
+    self.world.ui:playSound("bonusal2.wav")
```

Comment:
Potentially, we may not want to sound this if a wage request is auto granted. Thoughts?

Reply:
> Potentially, we may not want to sound this if a wage request is auto granted. Thoughts?

Sorry, I don't see any reason why we could ever want to silent this action. 
It was more looks like a bug and lack of consistency:

1. Take for example fire. playSound("sack.wav") is placed inside Staff:fire(). And looks like it doesn't consider the potential to not sound. 
2. Also It looks illogical that the sound of dismissal is inside the Staff functions and granting rise is not.  When calling Staff:fire() you need to think about sound. And when calling Staff:increaseWage() you need to think about sound. Looks like a lack of consistency.
3. So everywhere in the code when there were calls of increaseWage() there was a call of playSound("bonusal2.wav"). (Except that 'checkIfWaitedTooLongForRaise'). Now ("sack.wav") and ("bonusal2.wav") calls for staff only in one place.
4. I never saw in the original game both things without sound.

So, taking these arguments into account the idea of granting raises can happen without sound sounds strange. At least now. If we will need it for some reason, I believe it resolvable in both cases.

Reply:
I agree that it's slightly illogical to sound one and not the other. My thought path was if you turned auto-grant wage requests on you wouldn't really care when one is given out. However, I'm fine with either unless there's opinion elsewhere.